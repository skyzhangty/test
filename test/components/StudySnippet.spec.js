import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import StudySnippet from '../../src/scripts/components/StudySnippet';


describe('Study Snippet', () => {
	it('should render active', () => {
		const renderer = new ReactShallowRenderer();

		const study = {
			irbNum: 'irb',
			title: 'titlo',
			piFirstName: 'Robert',
			piMiddleName: '"the rock"',
			piLastName: 'McCormick',
			newVolunteers: 0,
			newMessages: 1,
			recommendations: 8,
			active: true,
		};

		const component  = renderer.render(
			<StudySnippet study={study}/>
		);

		expect(component).toMatchSnapshot();
	});

	it('should render inactive', () => {
		const renderer = new ReactShallowRenderer();

		const study = {
			irbNum: 'irb',
			title: 'titlo',
			piFirstName: 'Robert',
			piMiddleName: '"the rock"',
			piLastName: 'McCormick',
			newVolunteers: 0,
			newMessages: 1,
			recommendations: 8,
			active: false,
		};

		const component  = renderer.render(
			<StudySnippet study={study}/>
		);

		expect(component).toMatchSnapshot();
	});
});

