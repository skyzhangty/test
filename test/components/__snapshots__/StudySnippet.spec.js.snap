// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Study Snippet should render active 1`] = `
<li>
  <article
    className="study-snippet-st wow active"
  >
    <Connect(Link)
      bindings={
        Object {
          "irbNumber": "irb",
        }
      }
      to="studyHome"
    >
      <div
        className="banner"
      >
        <span
          aria-hidden="true"
          className="fa fa-check"
        />
        <span
          className="show-for-sr"
        >
          This study is
        </span>
        active
      </div>
      <div
        className="study-irb"
        id="divIRB-irb"
      >
        irb
      </div>
      <h1
        className="study-title"
        id="h1Title-irb"
      >
        titlo
      </h1>
      <div
        className="study-pi"
        id="divPI-irb"
      >
        McCormick, Robert "the rock"
      </div>
      <section
        className="study-notifications"
        id="sectionNotifications-irb"
      >
        <div>
          <span
            className="number"
          >
            1
          </span>
          Messages
        </div>
        <div>
          <span
            className="number"
          >
            8
          </span>
          Recommendations
        </div>
      </section>
    </Connect(Link)>
  </article>
</li>
`;

exports[`Study Snippet should render inactive 1`] = `
<li>
  <article
    className="study-snippet-st wow inactive"
  >
    <Connect(Link)
      bindings={
        Object {
          "irbNumber": "irb",
        }
      }
      to="studyHome"
    >
      <div
        className="banner"
      >
        <span
          aria-hidden="true"
          className="fa fa-times"
        />
        <span
          className="show-for-sr"
        >
          This study is
        </span>
        inactive
      </div>
      <div
        className="study-irb"
        id="divIRB-irb"
      >
        irb
      </div>
      <h1
        className="study-title"
        id="h1Title-irb"
      >
        titlo
      </h1>
      <div
        className="study-pi"
        id="divPI-irb"
      >
        McCormick, Robert "the rock"
      </div>
      <section
        className="study-notifications"
        id="sectionNotifications-irb"
      >
        <div>
          <span
            className="number"
          >
            1
          </span>
          Messages
        </div>
        <div>
          <span
            className="number"
          >
            8
          </span>
          Recommendations
        </div>
      </section>
    </Connect(Link)>
  </article>
</li>
`;
